doc/command_ref.rst

```diff
@@ -142,8 +142,41 @@ Options
 ``-h, --help``
     Show the help.
 
+.. _installroot-label:
+
 ``--installroot=<path>``
-    set install root
+    Specifies an alternative installroot, relative to where all packages will be
+    installed. Think of this like doing ``chroot <root> dnf`` except using
+    ``--installroot`` allows dnf to work before the chroot is created.
+- The followings are taken from or stored in installroot: *cachedir*, *gpgkey*
+  (they are imported into installroot from path described in .repo file that is
+  not related to installroot), *log files*, *releasever*.
+
+- The followings are searched first inside of installroot, and if they are not
+  resent, they are taken from host system: *config file* (unless
+  ``--config=<config file>`` command-line option is used, then the path is not
+  related to installroot), and :ref:`reposdir <reposdir-label>` (unless
+  ``--setopt=reposdir=<reposdir>`` command-line option is used, then the path is
+  not related to installroot).
```

Comment:
```
*config file* and :ref:`reposdir <reposdir-label>` are searched inside the installroot at first. If they are not resent, they are taken from host system.

Note:  When path is specified within command line argument (``--config=<config file>`` in case of *config file* and ``--setopt=reposdir=<reposdir>`` for *reposdir*) then this path is always relative to the host with no exceptions.
```